# Customer Behavior Analysis

---



---



Customer Behavior Analysis is a strategic process integral to product management, delving into the intricate ways customers engage with a product or service. By scrutinizing these interactions, product managers gain valuable insights to inform decision-making, customize strategies, and elevate overall customer experiences.

In [1]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

data = pd.read_csv("ecommerce_customer_data.csv")
print(data.head())

   User_ID  Gender  Age   Location Device_Type  Product_Browsing_Time  \
0        1  Female   23  Ahmedabad      Mobile                     60   
1        2    Male   25    Kolkata      Tablet                     30   
2        3    Male   32  Bangalore     Desktop                     37   
3        4    Male   35      Delhi      Mobile                      7   
4        5    Male   27  Bangalore      Tablet                     35   

   Total_Pages_Viewed  Items_Added_to_Cart  Total_Purchases  
0                  30                    1                0  
1                  38                    9                4  
2                  13                    5                0  
3                  20                   10                3  
4                  20                    8                2  


In [2]:
numeric_summary = data.describe()
print(numeric_summary)

          User_ID         Age  Product_Browsing_Time  Total_Pages_Viewed  \
count  500.000000  500.000000             500.000000          500.000000   
mean   250.500000   26.276000              30.740000           27.182000   
std    144.481833    5.114699              15.934246           13.071596   
min      1.000000   18.000000               5.000000            5.000000   
25%    125.750000   22.000000              16.000000           16.000000   
50%    250.500000   26.000000              31.000000           27.000000   
75%    375.250000   31.000000              44.000000           38.000000   
max    500.000000   35.000000              60.000000           50.000000   

       Items_Added_to_Cart  Total_Purchases  
count           500.000000       500.000000  
mean              5.150000         2.464000  
std               3.203127         1.740909  
min               0.000000         0.000000  
25%               2.000000         1.000000  
50%               5.000000         2.00

In [3]:
categorical_summary = data.describe(include='object')
print(categorical_summary)

       Gender Location Device_Type
count     500      500         500
unique      2        8           3
top      Male  Kolkata      Mobile
freq      261       71         178


In [4]:
fig = px.histogram(data, x='Age', title='Distribution of Age')
fig.show()

In [5]:
gender_counts = data['Gender'].value_counts().reset_index()
gender_counts.columns = ['Gender', 'Count']
fig = px.bar(gender_counts, x='Gender',
             y='Count',
             title='Gender Distribution')
fig.show()

## Analyzing Customer Behaviour

 let’s have a look at the relationship between the product browsing time and the total pages viewed:

In [6]:
fig = px.scatter(data, x='Product_Browsing_Time', y='Total_Pages_Viewed',
                 title='Product Browsing Time vs. Total Pages Viewed',
                 trendline='ols')
fig.show()

The scatter plot above tells us that there's no clear connection between the time people spend looking at products and how many pages they check out. It seems like customers don't necessarily explore more pages if they spend more time on the website. This could be because of different things like how the website is designed, if the content is interesting, or just personal preferences of each user.

## Let’s have a look at the Actions done by Gender:

In [7]:

gender_grouped = data.groupby('Gender')['Total_Pages_Viewed'].mean().reset_index()
gender_grouped.columns = ['Gender', 'Average_Total_Pages_Viewed']
fig = px.bar(gender_grouped, x='Gender', y='Average_Total_Pages_Viewed',
             title='Average Total Pages Viewed by Gender')
fig.show()

In [23]:

gender_grouped = data.groupby('Gender')['Items_Added_to_Cart'].mean().reset_index()
gender_grouped.columns = ['Gender', 'Items_Added_to_Cart']
fig = px.bar(gender_grouped, x='Gender', y='Items_Added_to_Cart',
             title='Average Items_Added_to_Cart by Male and Female')
fig.show()

In [24]:

gender_grouped = data.groupby('Gender')['Total_Purchases'].mean().reset_index()
gender_grouped.columns = ['Gender', 'Total_Purchases']
fig = px.bar(gender_grouped, x='Gender', y='Total_Purchases',
             title='Total_Purchases made by Male and Female')
fig.show()

## Let’s have a look at the Actions performed through different Devices

In [8]:
devices_grouped = data.groupby('Device_Type')['Total_Pages_Viewed'].mean().reset_index()
devices_grouped.columns = ['Device_Type', 'Average_Total_Pages_Viewed']
fig = px.bar(devices_grouped, x='Device_Type', y='Average_Total_Pages_Viewed',
             title='Average Total Pages Viewed by Devices')
fig.show()

In [17]:
devices_grouped = data.groupby('Device_Type')['Items_Added_to_Cart'].mean().reset_index()
devices_grouped.columns = ['Device_Type', 'Items_Added_to_Cart']
fig = px.bar(devices_grouped, x='Device_Type', y='Items_Added_to_Cart',
             title='Items_Added_to_Cart by different devices')
fig.show()

In [22]:
devices_grouped = data.groupby('Device_Type')['Total_Purchases'].mean().reset_index()
devices_grouped.columns = ['Device_Type', 'Total_Purchases']
fig = px.bar(devices_grouped, x='Device_Type', y='Total_Purchases',
             title='Total_Purchases by different devices')
fig.show()

## Let’s have a look at the Actions performed in different Locations

In [25]:
devices_grouped = data.groupby('Location')['Total_Pages_Viewed'].mean().reset_index()
devices_grouped.columns = ['Location', 'Average_Total_Pages_Viewed']
fig = px.bar(devices_grouped, x='Location', y='Average_Total_Pages_Viewed',
             title='Average Total Pages Viewed in different locations')
fig.show()

In [26]:
devices_grouped = data.groupby('Location')['Items_Added_to_Cart'].mean().reset_index()
devices_grouped.columns = ['Location', 'Items_Added_to_Cart']
fig = px.bar(devices_grouped, x='Location', y='Items_Added_to_Cart',
             title='Items_Added_to_Cart in different locations')
fig.show()

In [27]:
devices_grouped = data.groupby('Location')['Total_Purchases'].mean().reset_index()
devices_grouped.columns = ['Location', 'Total_Purchases']
fig = px.bar(devices_grouped, x='Location', y='Total_Purchases',
             title='Total_Purchases in different locations')
fig.show()

let’s calculate the customer lifetime value and visualize segments based on the customer lifetime value.

In [9]:
data['CLV'] = (data['Total_Purchases'] * data['Total_Pages_Viewed']) / data['Age']

data['Segment'] = pd.cut(data['CLV'], bins=[1, 2.5, 5, float('inf')],
                         labels=['Low Value', 'Medium Value', 'High Value'])

segment_counts = data['Segment'].value_counts().reset_index()
segment_counts.columns = ['Segment', 'Count']

fig = px.bar(segment_counts, x='Segment', y='Count',
             title='Customer Segmentation by CLV')
fig.update_xaxes(title='Segment')
fig.update_yaxes(title='Number of Customers')
fig.show()

let’s have a look at the churn rate of the customers:

In [10]:
data['Churned'] = data['Total_Purchases'] == 0

churn_rate = data['Churned'].mean()
print(churn_rate)

0.198


A churn rate of 0.198 signals a notable proportion of customers who have stopped using our product or service, emphasizing the urgency of addressing this issue. Effectively managing and reducing churn is crucial for sustaining business growth and ensuring ongoing profitability.